MapboxCoreNavigation/NavigationRouteOptions.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import MapboxDirections
+
+/**
+ A `NavigationRouteOptions` object is a structure that specifies the criteria for results returned by the Mapbox Directions API.
```

Comment:
We should briefly distinguish between this class and RouteOptions in the first sentence, for the benefit of developers looking at the code completion popup:

> A `NavigationRouteOptions` object specifies turn-by-turn-optimized criteria for results returned by the Mapbox Directions API.